src/browser/provider/pool.js

```diff
@@ -0,0 +1,145 @@
+import npm from 'npm';
```

Comment:
It would be really nice to get rid of this dep. First of all it's huge and npm programmatical API is unreliable.